### PR TITLE
Clean utilities/argparse and add missing tests

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

- Renames tests/utilities/argparse_utils.py to tests/utilities/argparse.py for consistency
- Adds missing tests to tests/utilities/argparse.py to boost coverage
- Removes duplicated code and unreached line from pytorch_lightning/utilities/argparse.py 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/utilities/test_argparse.py [hunk 1]
```diff
@@ -168,3 +202,13 @@ def test_add_argparse_args_no_argument_group():
     args = parser.parse_args(fake_argv)
     assert args.main_arg == "abc"
     assert args.my_parameter == 2
+
+
+def test_gpus_arg_default():
+    assert _gpus_arg_default('1,2') == '1,2'
```
Comment (Borda) [inline tests/utilities/test_argparse.py:208]: shall this be list `[1,2]`?

Comment (ethanwharris) [reply tests/utilities/test_argparse.py:208]: The argument has to be a string. Can change it to "[1,2]" if preferred, although the function just looks for a "," in the input.

File: tests/utilities/test_argparse.py [hunk 2]
```diff
@@ -1,17 +1,51 @@
...
+    _ = from_argparse_args(mock_trainer, parser)
+    mock_trainer.parse_argparser.assert_called_once_with(parser)
+
+
+def test_parse_argparser():
+    args = Namespace(a=1, b='test', c=None, d='not valid')
+    new_args = parse_argparser(ArgparseExample, args)
+    assert new_args.a == 1
+    assert new_args.b == 'test'
+    assert new_args.c
```
Comment (carmocca) [inline tests/utilities/test_argparse.py:45]: shouldn't this be?

```suggestion
    assert new_args.c is None
```

File: pytorch_lightning/utilities/argparse.py [hunk 3]
```diff
@@ -242,9 +242,6 @@ def add_argparse_args(
         if arg == 'track_grad_norm':
             use_type = float
 
-        if arg_default is inspect._empty:
-            arg_default = None
-
```
Comment (carmocca) [inline pytorch_lightning/utilities/argparse.py:247]: Why is this not needed anymore?

**Review by carmocca — COMMENTED** [review]
This does not fix any open issue, right?

Please do not ignore the "Before submitting" checklist

File: tests/utilities/test_argparse.py (hunk 2 above)
Comment (ethanwharris) [reply tests/utilities/test_argparse.py:45]: `new_args.c` should have the value `True`, so the equivalent would be `assert new_args.c is True` if preferred?

File: pytorch_lightning/utilities/argparse.py (hunk 3 above)
Comment (ethanwharris) [reply pytorch_lightning/utilities/argparse.py:247]: Those lines are never reached. The `allowed_types` filter above already avoids the `inspect._empty` case, so it is not possible to trigger the if.

**Comment by ethanwharris** [comment]
Hi @carmocca,

Sorry for missing the checklist. This doesn't close any open issues. I assumed an issue wouldn't be needed for just improving coverage. Feel free to close this PR if it needs further discussion :)

File: tests/utilities/test_argparse.py (hunk 2 above)
Comment (carmocca) [reply tests/utilities/test_argparse.py:45]: I didn't get why it should have the value `True` but it's explained here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/42a7b7058573bc659eb1fd6a64035ae80e270211/pytorch_lightning/utilities/argparse.py#L64-L72

**Review by carmocca — APPROVED** [review]
> Sorry for missing the checklist. This doesn't close any open issues. I assumed an issue wouldn't be needed for just improving coverage. Feel free to close this PR if it needs further discussion :)

Np! Just a friendly reminder.

LGTM!

File: tests/utilities/test_argparse.py (hunk 2 above)
Comment (ethanwharris) [reply tests/utilities/test_argparse.py:45]: Ah, that makes sense. I had forgotten about that :)

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Nice ! LGTM

**Pull request merged**